R/step-subset-slice.R

```diff
@@ -73,35 +74,38 @@ slice.dtplyr_step <- function(.data, ...) {
     i <- call2("{", assign_rows_var, subset_valid_rows)
   }
 
-  step_subset_i(.data, i)
+  step_subset_i(.data, i, by)
 }
 
 #' @rdname slice.dtplyr_step
 #' @importFrom dplyr slice_head
 #' @inheritParams dplyr::slice
 #' @export
-slice_head.dtplyr_step <- function(.data, ..., n, prop) {
+slice_head.dtplyr_step <- function(.data, ..., n, prop, by = NULL) {
   check_dots_empty()
+  by <- compute_by({{ by }}, .data, by_arg = "by", data_arg = ".data")
   size <- get_slice_size(n, prop, "slice_head")
   i <- expr(rlang::seq2(1L, !!size))
-  step_subset_i(.data, i = i)
+  step_subset_i(.data, i = i, by)
 }
 
 #' @rdname slice.dtplyr_step
 #' @importFrom dplyr slice_tail
 #' @export
-slice_tail.dtplyr_step <- function(.data, ..., n, prop) {
+slice_tail.dtplyr_step <- function(.data, ..., n, prop, by = NULL) {
   check_dots_empty()
+  by <- compute_by({{ by }}, .data, by_arg = "by", data_arg = ".data")
   size <- get_slice_size(n, prop, "slice_tail")
   i <- expr(rlang::seq2(.N - !!size + 1L, .N))
-  step_subset_i(.data, i = i)
+  step_subset_i(.data, i = i, by)
 }
 
 #' @rdname slice.dtplyr_step
 #' @importFrom dplyr slice_min
 #' @inheritParams dplyr::slice
 #' @export
-slice_min.dtplyr_step <- function(.data, order_by, ..., n, prop, with_ties = TRUE) {
+slice_min.dtplyr_step <- function(.data, order_by, ...,
```

Comment:
If the arguments no longer fit on one line, our style to put each argument on its own line. https://github.com/lionel-/codegrip makes it easy to do so.